File: 백준/Silver/11723. 집합/집합.cc
#include <bits/stdc++.h>
using namespace std;

int m, x;
string cmd;
int s = 0; // 비트마스크로 집합을 관리

int main()
{
    ios_base::sync_with_stdio(0), cin.tie(0), cout.tie(0);
    cin >> m;
    for (int i = 0; i < m; i++)
    {
        cin >> cmd;
        if (cmd == "add")
        {
            cin >> x;
            s |= (1 << (x - 1)); // x를 집합에 추가
        }
        else if (cmd == "remove")
        {
            cin >> x;
            s &= ~(1 << (x - 1)); // x를 집합에서 제거
        }
        else if (cmd == "check")
        {
            cin >> x;
            cout << ((s & (1 << (x - 1))) ? 1 : 0) << '\n'; // x가 집합에 있는지 확인
        }
        else if (cmd == "toggle")
        {
            cin >> x;
            s ^= (1 << (x - 1)); // x가 있으면 제거, 없으면 추가 (토글)
        }
        else if (cmd == "all")
        {
            s = (1 << 20) - 1; // {1, 2, ..., 20}으로 설정
        }
        else if (cmd == "empty")
        {
            s = 0; // 공집합으로 설정
        }
    }
    return 0;
}

File: 백준/Silver/11723. 집합/집합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

class Main {

    static int M, x;
    static boolean[] S;
    public static void main(String[] args) throws IOException {
        //1. 입력부
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        M = Integer.parseInt(br.readLine());
        S = new boolean[21];
        for (int i = 0; i < M; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String cmd = st.nextToken();
            switch(cmd) {
                case "add":
                    x = Integer.parseInt(st.nextToken());
                    S[x] = true;
                    break;
                case "remove":
                    x = Integer.parseInt(st.nextToken());
                    S[x] = false;
                    break;
                case "check":
                    x = Integer.parseInt(st.nextToken());
                    sb.append(S[x] ? 1 : 0).append('\n');
                    break;
                case "toggle":
                    x = Integer.parseInt(st.nextToken());
                    S[x] = !S[x];
                    break;
                case "all":
                    for (int j = 1; j <= 20; j++) {
                        S[j] = true;
                    }
                    break;
                case "empty":
                    for (int j = 1; j <= 20; j++) {
                        S[j] = false;
                    }
                    break;
            }
        }
        System.out.println(sb);
    }
}